flink-clients/src/main/java/org/apache/flink/client/program/ClusterClient.java

```diff
@@ -201,18 +157,15 @@ public JobSubmissionResult run(
 		int parallelism,
 		SavepointRestoreSettings savepointSettings) throws CompilerException, ProgramInvocationException {
 
-		OptimizedPlan optPlan = getOptimizedPlan(compiler, plan, parallelism);
```

Comment:
With this change `ClusterClient` doesn't have any usage of `compiler`. Also we want to hide `compiler` inside translator. So we can remove `Cluster.compiler` field.

Reply:
you're right, removing it.